### DISCUSSION TITLE
Integrating TUnit in other libraries

### DISCUSSION THREAD

**Discussion by dev-in-disguise** [thomhurst/TUnit#638] [General]
Hi!
I've seen that there has been efforts by you to introduce TUnit also in other testing libraries like FluentAssertions. 

While trying to migrate one of my work solutions to TUnit I noticed that we currently have a direct dependency on 2 other libraries that would require support for TUnit. 

The first one being AutoFixture (I opened a [discussion](https://github.com/AutoFixture/AutoFixture/discussions/1478) there how they would feel if someone would provide support for TUnit) and the second one being Specflow (though I'm heavily torn since that's basically discontinued and replaced by Reqnroll). 

My question now would be: should we as a community try to get TUnit support in as many libraries as we can or would you rather like to do that by yourself so that you have a better overview regarding dependencies that might need to be fixed with public API changes? Or do you say it would be wiser to wait with such things until you're at version 1 of TUnit since the API might still change heavily until then? 

Would love your feedback on this

**Comment by thomhurst** [comment]
I'm fine with the community asking to get the support you need - the more people that do the more likely it'll happen I reckon.

I'm happy to be tagged in threads and have an input or answer any questions if there is any.

Regarding the API, you're right, it may change, so might just be that their versioning remains pre-release while TUnits does too so both can make changes

**Reply by dev-in-disguise** [reply]
Alright! Thanks for the super fast reply! 

I personally would start by forking AutoFixture and implementing TUnit support now. In case the maintainer would not like to support TUnit should I publish and maintain the package myself or would you want to have this under a specific TUnit umbrella?

**Reply by thomhurst** [reply]
If autofixture don't want to include it themselves, it might make sense to add it to this repository. That way, if and when any API changes occur, then we'll get compile time errors so it'll force us to fix them. And the pipeline will pick up the project and publish a new nuget package along with all the other packages.

**Reply by dev-in-disguise** [reply]
Sounds like a great plan. Then I'll keep you in the loop regarding the maintainers decision

**Reply by AdaskoTheBeAsT** [reply]
Hi, I started working on integration with Reqnroll https://github.com/reqnroll/Reqnroll/pull/442
